src/sync/installation.ts

```diff
@@ -258,22 +262,11 @@ export const processInstallation =
 			};
 
 			const execute = async () => {
-				if (await booleanFlag(BooleanFlags.SIMPLER_PROCESSOR, true)) {
-					try {
-						return await processor(github, repository, cursor, 20);
-					} catch (err) {
-						logger.error({ err, job, github, repository, cursor, task }, "Error Executing Task");
-						handleGitHubError(err);
-					}
-				} else {
-					for (const perPage of [20, 10, 5, 1]) {
```

Comment:
We figured out the reason for this! Sometimes, the GraphQL API returns a "something went wrong with this query", with a suggestion to try again requesting a smaller sub graph. This was the solution to handle that error. When it fails with 20 items per page, try again with 10 and so on. 

This means we have to leave this logic in here (or rewrite it in some way).

This also makes apparent that there is a notable risk in switching to the GraphQL APIs everywhere, so I would very much suggest you split this PR up into multiple smaller PRs, and put the GraphQL change behind a feature flag so that we can do a slow rollout (or at least switch it back when we notice errors).

Reply:
yes, already discovered that and talked about it last week, which is why some of the graphql queries has less than 100 items per call to not overload graphql.  There needs to be an investigation as to why that is since graphql rules state we can have up to 500k nodes max in one call, but it seems to struggle with like 1000 nodes?

And there would be no way to feature flag this easily since the response would be too different between to two to be able to reuse the same code.  The service that calls the API returns different data, which means the transformer needs to change too, and a bunch of other things surrounding it.  If this coded was better abstracted, it would make sense, but this would increase the work by considerable amount.  

I hadn't noticed errors with my branch while testing it with the current queries and how they're setup.

Reply:
Hm, after some consideration, yeah, it could be a bit much in some instances.  I think the best approach is to try to keep 'chunks' at about 100 nodes each to reduce the amount of load on the github APIs.

Reply:
It's not just the number of nodes. I think the `MAX_NODE_LIMIT_EXCEEDED` error happens only rarely. But the GitHub GraphQL API returns a bunch of other errors on some queries as well.

We need to put the for loop back in to work around these errors: `for (const perPage of [20, 10, 5, 1])`. This loop is enabled in prod currently, so if you remove the `SIMPLER_PROCESSOR` feature flag, then make the default behavior use this loop, please.

Reply:
The loop had nothing to do with graphql as most of them were using rest and not graphql.  I'd imagine they used the loop specifically to get as much data as possible from the rest API before the rate limit hits.

I'd rather we try to fix it without using the loop as it's a horrible hack that shouldn't be used at all.  For those other graphql issues, we need to log/test/work around them.  From my experience, most of the issues come from having a query that takes too long, hence reducing the amount of nodes returned seems to fix it.